packages/frontend/src/App.tsx

```diff
@@ -20,20 +22,33 @@ function App() {
     <div className="app">
       <Routes>
         <Route element={<ProtectedPageLayout />}>
-          <Route path={PageRoutes.inbox} element={<Inbox key="inbox" viewType={'inbox'} />} />
-          <Route path={PageRoutes.partiesOverview} element={<PartiesOverviewPage key="partys" />} />
-          <Route path={PageRoutes.drafts} element={<Inbox key="draft" viewType={'drafts'} />} />
-          <Route path={PageRoutes.sent} element={<Inbox key="sent" viewType={'sent'} />} />
-          <Route path={PageRoutes.archive} element={<Inbox key="archive" viewType={'archive'} />} />
-          <Route path={PageRoutes.bin} element={<Inbox key="bin" viewType={'bin'} />} />
-          <Route path={PageRoutes.inboxItem} element={<DialogDetailsPage />} />
-          <Route path={PageRoutes.savedSearches} element={<SavedSearchesPage />} />
-          <Route path={PageRoutes.profile} element={<Profile />} />
-          <Route path={PageRoutes.authorize} element={<Authorize />} />
-          <Route path={PageRoutes.access} element={<Access />} />
-          <Route path={PageRoutes.notifications} element={<Notifications />} />
-          <Route path={PageRoutes.settings} element={<Settings />} />
-          <Route path={PageRoutes.activities} element={<Activities />} />
+          <Route
+            path={PageRoutes.inbox}
+            element={withErrorBoundary(<Inbox key="inbox" viewType={'inbox'} />, 'inbox')}
+          />
+          <Route
+            path={PageRoutes.partiesOverview}
+            element={withErrorBoundary(<PartiesOverviewPage key="partys" />, 'partiesOverview')}
+          />
+          <Route
```

Comment:
Have you considered determining which route failed checking window.location.pathname when the error is caught in`componentDidCatch` instead? Then you would know which route the error originated from, and could have a single, top-level ErrorBoundary instead of wrapping around each and every route.